From 8034802dbc7bf1aa332fb863e87b0cf58e09a4e8 Mon Sep 17 00:00:00 2001
From: Piyush Mehta <piyush.mehta@xilinx.com>
Date: Thu, 9 Apr 2020 23:46:20 +0530
Subject: [PATCH 1323/1851] usb: dwc3: Enable clock when USB resume occurs

commit f5179a0673b2a1b469b8e23eeec9ba7480cc653d from
https://github.com/Xilinx/linux-xlnx.git

USB clocks (which were disabled during suspend) need to
be enabled during the resume. This patch does the same.

Signed-off-by: Piyush Mehta <piyush.mehta@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/usb/dwc3/dwc3-of-simple.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/dwc3/dwc3-of-simple.c b/drivers/usb/dwc3/dwc3-of-simple.c
index 49f7edcff395..bac775a9cc27 100644
--- a/drivers/usb/dwc3/dwc3-of-simple.c
+++ b/drivers/usb/dwc3/dwc3-of-simple.c
@@ -664,10 +664,15 @@ static int __maybe_unused dwc3_of_simple_suspend(struct device *dev)
 static int __maybe_unused dwc3_of_simple_resume(struct device *dev)
 {
 	struct dwc3_of_simple *simple = dev_get_drvdata(dev);
+	int ret;
 
 	if (simple->wakeup_capable || simple->dwc->is_d3)
 		return 0;
 
+	ret = clk_bulk_enable(simple->num_clocks, simple->clks);
+	if (ret)
+		return ret;
+
 	if (simple->need_reset)
 		reset_control_deassert(simple->resets);
 
-- 
2.31.1

